apps/app_laplacian.cc

```diff
@@ -33,10 +33,12 @@ main(int argc, char *argv[])
         prm_file = "parameters.prm";
       if (prm_file.find("23d") != std::string::npos)
         {
+#ifndef MATRIX_FREE_PATH
           ProblemParameters<2, 3> par;
           PoissonProblem<2, 3>    problem(par);
```

Comment:
We have to guard the codim-1 cases if we want to use matrix-free in the same file.

Reply:
The codimension one case does not work?

Reply:
Yes, it does not work. The matrix-free evauators do not support it

Reply:
How much would it take to support it? I think we need the co-dimension one case for the B and BT operators in the actual application of the vessels.

Reply:
$B$ and $B^T$ work in the co-dimension one case (there we just need to do point evaluations). 

What is not supported in co-dimension one case is the action of $A$ (grad-grad)... but that term is always volumetric.

Reply:
To be more specific: the `MatrixFree::reinit()` function (https://dealii.org/developer/doxygen/deal.II/classMatrixFree.html#adb324d469b296f2a6921b41208732ddf) has only one template argument associated to the space dimension. In our case, the mixed-dimensional coupling is occurring through $B$ and $B^T$ which are based on `RemotePointEvaluation` and `FEPointEvaluation`, which supports co-dimension one evaluations.

Reply:
Then the guard is not necessary. We do not actually compute the grad-grad in codimension one. The codimension is only intended for the immersed problem, not for the outer problem

Reply:
Fixed in last commit. The guard was necessary to compile the code (there were some dim-spacedim issues when matrix-free grad-grad components were added, my bad).